Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -91,24 +107,35 @@ void GridMasonryLayout::addItemsToFirstTrack(const HashMap<RenderBox*, GridArea>
         insertIntoGridAndLayoutItem(item, gridArea);
 }
 
+void GridMasonryLayout::placeItemsUsingOrderModifiedDocumentOrder()
+{
+    for (auto* child : m_itemsWithDefiniteGridAxisPosition) {
+        if (hasDefiniteGridAxisPosition(child, gridAxisDirection()))
+            insertIntoGridAndLayoutItem(child, positionForDefiniteGridAxisItem(*child));
+        else
+            insertIntoGridAndLayoutItem(child, positionForIndefiniteGridAxisItem(child));
+    }   
+}
+
 void GridMasonryLayout::placeItemsWithDefiniteGridAxisPosition(const Vector<RenderBox*>& itemsWithDefiniteGridAxisPosition)
 {
     for (auto* item : itemsWithDefiniteGridAxisPosition) {
-        auto itemSpan = m_renderGrid.currentGrid().gridItemSpan(*item, gridAxisDirection());
-
-        ASSERT(!itemSpan.isIndefinite());
-
-        itemSpan.translate(m_renderGrid.currentGrid().explicitGridStart(gridAxisDirection()));
-
-        GridArea area = m_masonryAxisDirection == ForRows ? GridArea(m_masonryAxisSpan, itemSpan) : GridArea(itemSpan, m_masonryAxisSpan);
-        insertIntoGridAndLayoutItem(item, area);
+        insertIntoGridAndLayoutItem(item, positionForDefiniteGridAxisItem(*item));
     }
 }
 
+GridArea GridMasonryLayout::positionForDefiniteGridAxisItem(const RenderBox& child) const
```

Comment:
Nit: The name of this method makes me think it returns a position (e.g., [x,y] location) of some kind, but it returns an area (e.g, [width, height], or possibly [x, y, width, height]). Perhaps this is the nomenclature used throughout Grid code, but I was slightly confused while reading.

Reply:
I don't think position here refers to a specific grid concept so this could be changed to something like `gridAreaForDefiniteGridAxisItem`

CC @mattwoodrow in case I am wrong on this though

Reply:
Ended up changing the names because I agree it probably it more intuitive